### PR TITLE
Mark some resource metadata as required.

### PR DESCRIPTION
**Pull request by cbookg**
workspaceId, resourceId, name and types should always be populated.

### CODE REVIEW

**Review by yuhuyoyo — APPROVED** [review]

**Review by melissachang — APPROVED** [review]

**Review by ddietterich — CHANGES_REQUESTED** [review]
This looks like an incompatible change.
What happens to Rawls when this goes in? Will it break their code???

We have to be very careful changing the API.

**Comment by ddietterich** [comment]
> I'm expecting this to only be a change to an "output" field

When I saw the PR with two approves and potential problems I flagged it. Now that I have looked more closely at the code, I agree that it is only on the output side and I don't think it will cause issues with Scala or Java.

I am unhappy that the file is reformatted. We have dueling formatters and space-equivalent changes keep flip-flopping. Would it be possible to make your change without the reformatting?

File: openapi/src/common/schemas.yaml [hunk 1]
```diff
@@ -225,6 +225,7 @@ components:
     # All resource objects include this resource metadata object and call it 'metadata'
     ResourceMetadata:
       type: object
+      required: [workspaceId, resourceId, name, resourceType, stewardshipType]
```
[the remark this answers is not in the document]
Comment (cbookg) [reply openapi/src/common/schemas.yaml:228]: leaving

**Comment by cbookg** [comment]
> I am unhappy that the file is reformatted. We have dueling formatters and space-equivalent changes keep flip-flopping. Would it be possible to make your change without the reformatting?

Reverted all of the other lines.

Note the current file is inconsistently using both forms, but I agree that should be fixed in a distinct PR.

**Review by ddietterich — APPROVED** [review]
Thanks! I'll see if there is a spotless-style formatter that will keep all of the YAML parts consistent.

**Pull request merged**